spec/instance_topic_fields_spec.rb

```diff
@@ -0,0 +1,70 @@
+RSpec.describe SparqlToSwSolr::InstanceSolrDoc::InstanceTopicFields do
+
+  let(:instance_uri) { 'http://ld4p-test.stanford.edu/1234567890#Instance' }
+  let(:isd) { SparqlToSwSolr::InstanceSolrDoc.new(instance_uri) }
+
+  context 'SPARQL' do
+    let(:sparql_conn) { double('sparql client') }
+    let(:solutions) { RDF::Query::Solutions.new }
+
+    before(:each) do
+      allow(isd).to receive(:sparql).and_return(sparql_conn)
+      expect(sparql_conn).to receive(:query).and_return(solutions)
+    end
+
+    shared_examples 'it_indexes_topics' do
+      # examples calling this shared_examples must declare `let` for:
+      # :solr_field
+      let(:doc_hash) { isd.send(:add_doc_topic_fields, {}) }
+      let(:solr_topics) { doc_hash[solr_field] }
+      let(:bf_topics) { ['math', 'physics'] }
+
+      it 'empty Array when there are no topics' do
+        expect(solr_topics).to be_an Array
+        expect(solr_topics).to be_empty
+      end
+      it 'is an Array - multi-valued Solr field - when there is one topic' do
+        solutions << RDF::Query::Solution.new(p: 'madsrdf:authoritativeLabel', topicLabel: bf_topics.first)
+        expect(solr_topics).to be_an Array
+        expect(solr_topics).to include bf_topics.first
+      end
+      it 'is an Array - multi-valued Solr field - when there are multiple topics' do
+        solutions << RDF::Query::Solution.new(p: 'madsrdf:authoritativeLabel', topicLabel: bf_topics.first)
+        solutions << RDF::Query::Solution.new(p: 'madsrdf:authoritativeLabel', topicLabel: bf_topics.last)
+        expect(solr_topics).to be_an Array
+        expect(solr_topics).to include bf_topics.first
```

Comment:
I had to hunt around to ensure that bf_topics.first != bf_topics.last.   Also note that the facet field should preserve case, so it might be better to use values cribbed from actual data.  (Like, the actual data returned by sparql is `Conductors (Music)--Italy--Biography.` and topic_facet should be `Conductors (Music)`, and topic_search should be the whole string.   I would look for values with no `--`, with one `--` and more than one `--` as the code should gracefully handle all those cases.  You sort of did that below.